app/controllers/provider_interface/provider_users_invitations_controller.rb

```diff
@@ -157,5 +161,9 @@ def setup_permission_form
         permissions: @wizard.permissions_for_provider(@provider.id),
       )
     end
+
+    def persistence_key_for_current_user
+      "provider_user_invitation_wizard-#{current_provider_user.id}"
```

Comment:
It's not a big deal, but the controller shouldn't have to worry about generating keys. We have `STATE_STORE_KEY` in the wizard, and it'd be better if the wizard decides how to generate its own keys. Having said that, the wizard has no access to `current_user` currently.

We could start passing a user context to the wizard, but I think a better approach would be to pass `session:` to both wizards, making the two store implementations have a uniform API, making them truly interchangeable. The session object exposes an `.id` which is unique to the browser session. How would you feel about using a combination of `STATE_STORE_KEY` and `session.id` as the key instead? `session.id` can be `nil`, but this happens only when a browser first visits the site or if it rejects cookies and is guaranteed to be initialised after authentication.

Reply:
This is a nice idea, thank you. Will have a go.

Reply:
I've had a go, and had a think, and I think I prefer the current implementation for the following reasons:

- It's simply more complicated to have the state store depend on a session object than a string, and it's even a bit weird when the state store has nothing to do with sessions in the first place
- It feels more logical to have the store itself,  rather than some ingredient of the store, as a dependency of the wizard
- I don't think having equivalent constructors is worth the trouble of taking a more complicated dependency

Reply:
The current implementation works fine so I'm happy to go with it. There is another side-effect to applying `user_id` as part of the wizard key, wizard state will be persisted across browsers, incognito sessions or even machines (up to the 24h expiry). Using a browser session id would probably be more in line with user expectations, but it's not that important.